fluster/decoders/gstreamer.py

```diff
@@ -37,18 +38,33 @@ class GStreamer(Decoder):
     api = None
     provider = None
     name = None
+    sink = None
 
     def __init__(self):
         super().__init__()
         if not self.name:
             self.name = f'{self.provider}-{self.codec.value}-{self.api}-Gst{self.gst_api}'
         self.description = f'{self.provider} {self.codec.value} {self.api} decoder for GStreamer {self.gst_api}'
         self.cmd = normalize_binary_cmd(self.cmd)
+        gst_version = self.get_gst_version()
+        if gst_version >= '1.20.0':
+            self.sink = 'videoconformancesink'
+        else:
+            self.sink = 'filesink'
+
+    def get_gst_version(self):
```

Comment:
Could you please add the return type here? `-> str` 
I know not every method has it, but we're trying to be more explicit with new code so that the linter warns us if there's something catchy.

Apart from that. I think we could take out this out of the class as a free function and use `lru_cache` to prevent spawning a new process for every single GStreamer-based decoder. Maybe we can even move it to `utils.py`, but since it's GStreamer-related, we could as well leave it here. @rgonzalezfluendo, @ndufresne what do you think?

Reply:
Sure, I'll read-up about lru_cache, I'm not doing much python these days. I notice that it was endlessly spawning a process, which was fairly annoying. Note that I'm on hold upstream gstreamer, I think no one really understand our use case, they all expected fluster to get a MD5 from gstreamer, not a raw yuv file.

Reply:
My two cents: Why not check if `videoconformancesink` exists instead of check the gst version? 

It would work with running `gst-inspect-1.0 videoconformancesink`

Reply:
Good point, that would work too, I see it returns 255 if you try and inspect a missing feature. Ah, even better, gst-inspect-1.0 --exist. Will update.